File: leetcode/rust-leetcode/src/198.house-robber.rs
use std::cmp::max;
impl Solution {
    pub fn rob(mut nums: Vec<i32>) -> i32 {
        if nums.len() < 3 {
            return *nums.iter().max().unwrap();
        }
        nums[2] += nums[0];
        let mut m = max(max(nums[0], nums[1]), nums[2]);
        for i in 3..nums.len() {
            nums[i] += max(nums[i-2], nums[i-3]);
            m = m.max(nums[i]);
        }
        m
    }
}

File: leetcode/python/198.house-robber.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        if len(nums) < 3:
            return max(nums)
        nums[2] += nums[0]
        for i in range(3, len(nums)):
            nums[i] += max(nums[i-2], nums[i-3])
        return max(nums[-2:])

